src/components/menu/menu-item/menu-item.d.ts

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { IconTypes } from '../../../utils/helpers/options-helper/options-helper';
+
+export interface MenuItemProps {
+  children?: React.ReactNode;
+  onClick?: () => void;
```

Comment:
The onClick needs to have the argument type. Without it it will cause issues for Typescript projects https://github.com/Sage/carbon/issues/3143